### ISSUE TITLE
when editing the video.. in timeline recyclerview just drag forward and backwards fast and it crashes

### ISSUE THREAD

**Issue by rocky8roy** [wlanjie/trinity#96]
**Question:** when editing the video.. in timeline recyclerview just drag forward and backwards fast and it crashes
**System version:** 10.0

**Mobile phone brands:** LG G7

**LogFile:** I/ViewRootImpl: ViewRoot's Touch Event : ACTION_DOWN
D/com.trinity.sample.view.ThumbLineBar: ScrollStateChanged SCROLL_STATE_DRAGGING
D/com.trinity.sample.view.ThumbLineBar: -------------- pause --------------
E/trinity: [player.cc, Seek, 309]:enter: Seek seek: 23415 index: 0
I/trinity: [message_queue.cc, Flush, 75]:
     Player Message Queue Flush .... and this time the queue_ Size is 0 
    [player.cc, Seek, 327]:leave: Seek
E/trinity: [player.cc, HandleMessage, 825]:enter message seek index: 0 seek_index: 0 time: 23415
E/trinity: [av_play.c, av_play_seek, 809]:enter: av_play_seek seek_to: 23415 pre_seek: 23266 duration: 23691000 frame_size; 4 max_size; 4 pd: 4020 pmd: 120000 vc: 4
    [av_play.c, change_status, 1043]:change_status: 3
D/com.trinity.sample.view.ThumbLineBar: -------------- pause --------------
E/trinity: [player.cc, Seek, 309]:enter: Seek seek: 23140 index: 0
I/trinity: [message_queue.cc, Flush, 75]:
     Player Message Queue Flush .... and this time the queue_ Size is 0 
    [player.cc, Seek, 327]:leave: Seek
E/trinity: [player.cc, HandleMessage, 825]:enter message seek index: 0 seek_index: 0 time: 23140
E/trinity: [av_play.c, av_play_seek, 809]:enter: av_play_seek seek_to: 23140 pre_seek: 23415 duration: 23691000 frame_size; 4 max_size; 4 pd: 3019 pmd: 120000 vc: 3
E/trinity: [av_play.c, flush_packet_queue, 566]:enter: flush_packet_queue
    [av_play.c, change_status, 1043]:change_status: 3
E/trinity: [av_play.c, read_thread, 582]:read seeking
    [av_play.c, clear_frame_queue, 409]:enter: clear_frame_queue
E/trinity: [av_play.c, video_decode_hw_thread, 513]:packet == &context->video_packet_queue->flush_packet
A/libc: Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0xbf in tid 31729 (Thread-34046), pid 30278 (.trinity.sample)
E/trinity: [player.cc, RenderFrameBuffer, 1169]:av_play_frame: 0x0 count: 0
D/com.trinity.sample.view.ThumbLineBar: -------------- pause --------------
E/trinity: [player.cc, Seek, 309]:enter: Seek seek: 22928 index: 0
I/trinity: [message_queue.cc, Flush, 75]:
     Player Message Queue Flush .... and this time the queue_ Size is 0 
    [player.cc, Seek, 327]:leave: Seek
E/trinity: [player.cc, HandleMessage, 825]:enter message seek index: 0 seek_index: 0 time: 22928
    [av_play.c, av_play_seek, 809]:enter: av_play_seek seek_to: 22928 pre_seek: 23140 duration: 23691000 frame_size; 0 max_size; 4 pd: 0 pmd: 120000 vc: 0
    [av_play.c, flush_packet_queue, 566]:enter: flush_packet_queue
    [av_play.c, change_status, 1043]:change_status: 3
E/trinity: [av_play.c, video_decode_hw_thread, 513]:packet == &context->video_packet_queue->flush_packet
D/com.trinity.sample.view.ThumbLineBar: -------------- pause --------------
E/trinity: [player.cc, Seek, 309]:enter: Seek seek: 22822 index: 0
I/trinity: [message_queue.cc, Flush, 75]:
     Player Message Queue Flush .... and this time the queue_ Size is 0 
    [player.cc, Seek, 327]:leave: Seek
E/trinity: [player.cc, HandleMessage, 825]:enter message seek index: 0 seek_index: 0 time: 22822
    [av_play.c, av_play_seek, 809]:enter: av_play_seek seek_to: 22822 pre_seek: 22928 duration: 23691000 frame_size; 1 max_size; 4 pd: 0 pmd: 120000 vc: 0
    [av_play.c, flush_packet_queue, 566]:enter: flush_packet_queue
E/trinity: [av_play.c, change_status, 1043]:change_status: 3
D/com.trinity.sample.view.ThumbLineBar: -------------- pause --------------
E/trinity: [player.cc, Seek, 309]:enter: Seek seek: 22759 index: 0
I/trinity: [message_queue.cc, Flush, 75]:
     Player Message Queue Flush .... and this time the queue_ Size is 0 
    [player.cc, Seek, 327]:leave: Seek
E/trinity: [player.cc, HandleMessage, 825]:enter message seek index: 0 seek_index: 0 time: 22759
    [av_play.c, av_play_seek, 809]:enter: av_play_seek seek_to: 22759 pre_seek: 22822 duration: 23691000 frame_size; 1 max_size; 4 pd: 0 pmd: 120000 vc: 1
D/com.trinity.sample.view.ThumbLineBar: -------------- pause --------------
E/trinity: [av_play.c, flush_packet_queue, 566]:enter: flush_packet_queue
E/trinity: [player.cc, Seek, 309]:enter: Seek seek: 22357 index: 0
E/trinity: [av_play.c, change_status, 1043]:change_status: 3
I/trinity: [message_queue.cc, Flush, 75]:
     Player Message Queue Flush .... and this time the queue_ Size is 0 
    [player.cc, Seek, 327]:leave: Seek
E/trinity: [player.cc, HandleMessage, 825]:enter message seek index: 0 seek_index: 0 time: 22357
    [av_play.c, av_play_seek, 809]:enter: av_play_seek seek_to: 22357 pre_seek: 22759 duration: 23691000 frame_size; 1 max_size; 4 pd: 0 pmd: 120000 vc: 1
    [av_play.c, flush_packet_queue, 566]:enter: flush_packet_queue
    [av_play.c, change_status, 1043]:change_status: 3
E/trinity: [av_play.c, video_decode_hw_thread, 513]:packet == &context->video_packet_queue->flush_packet
D/com.trinity.sample.view.ThumbLineBar: -------------- pause --------------
E/trinity: [player.cc, Seek, 309]:enter: Seek seek: 22315 index: 0
I/trinity: [message_queue.cc, Flush, 75]:
     Player Message Queue Flush .... and this time the queue_ Size is 0 
I/trinity: [player.cc, Seek, 327]:leave: Seek
E/trinity: [player.cc, HandleMessage, 825]:enter message seek index: 0 seek_index: 0 time: 22315
    [av_play.c, av_play_seek, 809]:enter: av_play_seek seek_to: 22315 pre_seek: 22357 duration: 23691000 frame_size; 1 max_size; 4 pd: 0 pmd: 120000 vc: 0
    [av_play.c, flush_packet_queue, 566]:enter: flush_packet_queue
    [av_play.c, change_status, 1043]:change_status: 3
E/trinity: [av_play.c, video_decode_hw_thread, 513]:packet == &context->video_packet_queue->flush_packet
D/com.trinity.sample.view.ThumbLineBar: -------------- pause --------------
E/trinity: [player.cc, Seek, 309]:enter: Seek seek: 22209 index: 0
I/trinity: [message_queue.cc, Flush, 75]:
     Player Message Queue Flush .... and this time the queue_ Size is 0 
    [player.cc, Seek, 327]:leave: Seek
E/trinity: [player.cc, HandleMessage, 825]:enter message seek index: 0 seek_index: 0 time: 22209
    [av_play.c, av_play_seek, 809]:enter: av_play_seek seek_to: 22209 pre_seek: 22315 duration: 23691000 frame_size; 1 max_size; 4 pd: 0 pmd: 120000 vc: 0
    [av_play.c, flush_packet_queue, 566]:enter: flush_packet_queue
    [av_play.c, change_status, 1043]:change_status: 3
E/trinity: [av_play.c, video_decode_hw_thread, 513]:packet == &context->video_packet_queue->flush_packet
D/com.trinity.sample.view.ThumbLineBar: -------------- pause --------------
E/trinity: [player.cc, Seek, 309]:enter: Seek seek: 22145 index: 0
I/trinity: [message_queue.cc, Flush, 75]:
     Player Message Queue Flush .... and this time the queue_ Size is 0 
    [player.cc, Seek, 327]:leave: Seek
E/trinity: [player.cc, HandleMessage, 825]:enter message seek index: 0 seek_index: 0 time: 22145
    [av_play.c, av_play_seek, 809]:enter: av_play_seek seek_to: 22145 pre_seek: 22209 duration: 23691000 frame_size; 1 max_size; 4 pd: 0 pmd: 120000 vc: 0
    [av_play.c, flush_packet_queue, 566]:enter: flush_packet_queue
    [av_play.c, change_status, 1043]:change_status: 3
E/trinity: [av_play.c, video_decode_hw_thread, 513]:packet == &context->video_packet_queue->flush_packet
D/com.trinity.sample.view.ThumbLineBar: -------------- pause --------------
E/trinity: [player.cc, Seek, 309]:enter: Seek seek: 21870 index: 0
I/trinity: [message_queue.cc, Flush, 75]:
     Player Message Queue Flush .... and this time the queue_ Size is 0 
    [player.cc, Seek, 327]:leave: Seek
E/trinity: [player.cc, HandleMessage, 825]:enter message seek index: 0 seek_index: 0 time: 21870
    [av_play.c, av_play_seek, 809]:enter: av_play_seek seek_to: 21870 pre_seek: 22145 duration: 23691000 frame_size; 1 max_size; 4 pd: 0 pmd: 120000 vc: 0
    [av_play.c, flush_packet_queue, 566]:enter: flush_packet_queue
    [av_play.c, change_status, 1043]:change_status: 3
D/com.trinity.sample.view.ThumbLineBar: -------------- pause --------------
E/trinity: [player.cc, Seek, 309]:enter: Seek seek: 21828 index: 0
I/trinity: [message_queue.cc, Flush, 75]:
     Player Message Queue Flush .... and this time the queue_ Size is 0 
    [player.cc, Seek, 327]:leave: Seek
E/trinity: [player.cc, HandleMessage, 825]:enter message seek index: 0 seek_index: 0 time: 21828
    [av_play.c, av_play_seek, 809]:enter: av_play_seek seek_to: 21828 pre_seek: 21870 duration: 23691000 frame_size; 1 max_size; 4 pd: 0 pmd: 120000 vc: 1
    [av_play.c, flush_packet_queue, 566]:enter: flush_packet_queue
    [av_play.c, change_status, 1043]:change_status: 3
E/trinity: [av_play.c, video_decode_hw_thread, 513]:packet == &context->video_packet_queue->flush_packet
D/com.trinity.sample.view.ThumbLineBar: -------------- pause --------------
E/trinity: [player.cc, Seek, 309]:enter: Seek seek: 21807 index: 0
I/trinity: [message_queue.cc, Flush, 75]:
     Player Message Queue Flush .... and this time the queue_ Size is 0 
    [player.cc, Seek, 327]:leave: Seek
E/trinity: [player.cc, HandleMessage, 825]:enter message seek index: 0 seek_index: 0 time: 21807
    [av_play.c, av_play_seek, 809]:enter: av_play_seek seek_to: 21807 pre_seek: 21828 duration: 23691000 frame_size; 1 max_size; 4 pd: 0 pmd: 120000 vc: 0
    [av_play.c, flush_packet_queue, 566]:enter: flush_packet_queue
    [av_play.c, change_status, 1043]:change_status: 3
E/trinity: [av_play.c, video_decode_hw_thread, 513]:packet == &context->video_packet_queue->flush_packet
D/com.trinity.sample.view.ThumbLineBar: -------------- pause --------------
E/trinity: [player.cc, Seek, 309]:enter: Seek seek: 21786 index: 0
I/trinity: [message_queue.cc, Flush, 75]:
     Player Message Queue Flush .... and this time the queue_ Size is 0 
    [player.cc, Seek, 327]:leave: Seek
E/trinity: [player.cc, HandleMessage, 825]:enter message seek index: 0 seek_index: 0 time: 21786
    [av_play.c, av_play_seek, 809]:enter: av_play_seek seek_to: 21786 pre_seek: 21807 duration: 23691000 frame_size; 1 max_size; 4 pd: 0 pmd: 120000 vc: 0
    [av_play.c, flush_packet_queue, 566]:enter: flush_packet_queue
    [av_play.c, change_status, 1043]:change_status: 3
E/trinity: [av_play.c, video_decode_hw_thread, 513]:packet == &context->video_packet_queue->flush_packet
I/trinity: [player.cc, Pause, 341]:Player Pause
I/trinity: [player.cc, HandleMessage, 961]:enter kPlayerPause
    [player.cc, HandleMessage, 973]:leave kPlayerPause
Connected to process 32133 on device 'lge-lm_g710-LMG710EAW2687f7d2'.
Capturing and displaying logcat messages from application. This behavior can be disabled in the "Logcat output" section of the "Debugger" settings page.
I/Typeface: Current DefaultTypeface ( default_roboto )
W/Glide: Failed to find GeneratedAppGlideModule. You should include an annotationProcessor compile dependency on com.github.bumptech.glide:compiler in your application and a @GlideModule annotated AppGlideModule implementation or LibraryGlideModules will be silently ignored
D/PhoneWindow: windowLightStatusBar : false, disable View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR
W/.trinity.sampl: Accessing hidden method Landroid/view/View;->computeFitSystemWindows(Landroid/graphics/Rect;Landroid/graphics/Rect;)Z (greylist, reflection, allowed)
W/.trinity.sampl: Accessing hidden method Landroid/view/ViewGroup;->makeOptionalFitsSystemWindows()V (greylist, reflection, allowed)
I/SensorManager: registerListenerImpl() [Sensor: LGE Accelerometer Non-wakeup, Rate: 200000, SensorEventListener: android.view.OrientationEventListener$SensorEventListenerImpl@fd31c46] by android.view.OrientationEventListener.enable():93
I/TouchFilterInputStageHelper: [needTouchEventFilter] Displayid : 0
D/BezellessGripSuppressionFilter: getdisplaysize, x : 1440 y : 3120
I/AdrenoGLES: QUALCOMM build                   : d424b94, If195280fd4
    Build Date                       : 12/24/19
    OpenGL ES Shader Compiler Version: EV031.27.05.03
    Local Branch                     : 
    Remote Branch                    : quic/gfx-adreno.lnx.1.0.r72-rel
    Remote Branch                    : NONE
    Reconstruct Branch               : NOTHING
    Build Config                     : S P 8.0.12 AArch32
I/AdrenoGLES: PFP: 0x016ee187, ME: 0x00000000
I/trinity: [camera_record.cc, PrepareEGLContext, 87]:enter PrepareEGLContext
    [camera_record.cc, PrepareEGLContext, 128]:leave PrepareEGLContext
I/trinity: [camera_record.cc, ProcessMessage, 516]:enter ProcessMessage
I/trinity: [camera_record.cc, SetRenderType, 145]:SetRenderType: 1
I/trinity: [camera_record.cc, Initialize, 551]:enter Initialize
I/trinity: [camera_record.cc, SetFrame, 152]:SetFrame: 2
I/trinity: [egl_core.cc, Init, 125]:enter: Init
D/tag: [, , 0]:startPreview
I/trinity: [camera_record.cc, ResetRenderSize, 166]:ResetRenderSize screen_width: 1440 screen_height: 2858
I/trinity: [egl_core.cc, Init, 161]:leave: Init
    [camera_record.cc, CreatePreviewSurface, 625]:enter CreatePreviewSurface
    [camera_record.cc, onSurfaceCreated, 388]:enter onSurfaceCreated
I/System.out: onSurfaceCreated
I/trinity: [camera_record.cc, onSurfaceCreated, 408]:leave onSurfaceCreated
    [camera_record.cc, CreatePreviewSurface, 634]:leave CreatePreviewSurface
    [camera_record.cc, StartCameraPreview, 342]:enter StartCameraPreview
    [, , 0]:enter startPreviewFromNative
I/trinity: [, , 0]:request preview width: 1280 height: 720
W/Gralloc3: mapper 3.x is not supported
I/Timeline: Timeline: Activity_idle id: android.os.BinderProxy@73dec7b time:74152079
I/trinity: [, , 0]:setPreviewSize width: 1280 height: 720
I/trinity: [, , 0]:leave startPreviewFromNative
I/trinity: [camera_record.cc, StartCameraPreview, 362]:leave StartCameraPreview
    [camera_record.cc, ConfigCamera, 304]:enter ConfigCamera
I/trinity: [camera_record.cc, ConfigCamera, 338]:leave ConfigCamera
W/Gralloc3: allocator 3.x is not supported
I/trinity: [opengl.cc, CompileShader, 302]:#ifdef GL_ES                                                                           
    precision highp float;                                                                 
    #endif                                                                                 
    attribute vec4 position;                                                               
    attribute vec4 inputTextureCoordinate;                                                 
    varying vec2 textureCoordinate;                                                        
    void main() {                                                                          
        gl_Position = position;                                                            
        textureCoordinate = inputTextureCoordinate.xy;                                     
    }                                                                                      
I/trinity: [opengl.cc, CompileShader, 302]:#ifdef GL_ES                                                                           
    precision highp float;                                                                 
    #endif                                                                                 
    varying vec2 textureCoordinate;                                                        
    uniform sampler2D inputImageTexture;                                                   
    void main() {                                                                          
        gl_FragColor = texture2D(inputImageTexture, textureCoordinate);                    
    }                                                                                      
    [camera_record.cc, Initialize, 571]:leave Initialize
I/trinity: [opengl.cc, CompileShader, 302]:attribute vec4 position;                                                               
    attribute vec4 inputTextureCoordinate;                                                 
    varying vec2 textureCoordinate;                                                        
    uniform highp mat4 textureMatrix;                                                      
    void main() {                                                                          
      textureCoordinate = (textureMatrix * inputTextureCoordinate).xy;                     
      gl_Position = position;                                                              
    }                                                                                      
    [opengl.cc, CompileShader, 302]:#extension GL_OES_EGL_image_external : require                                         
    precision mediump float;                                                               
    uniform samplerExternalOES inputImageTexture;                                              
    varying vec2 textureCoordinate;                                                        
    void main() {                                                                          
      gl_FragColor = texture2D(inputImageTexture, textureCoordinate);                          
    }                                                                                      
E/trinity: [opengl.cc, SetFrame, 136]:SetFrame source_width: 720 source_height: 1280 target_width: 1440 target_height: 2858
    [opengl.cc, SetFrame, 159]:scale_width: 1.116406 scale_height: 1.000000
I/System.out: onPause
I/trinity: [, , 0]:destroySurface
    [camera_record.cc, DestroyEGLContext, 173]:DestroyEGLContext enter
I/trinity: [camera_record.cc, Destroy, 576]:enter Destroy
I/trinity: [opengl.cc, ~OpenGL, 87]:enter: ~OpenGL program: 4
    [opengl.cc, ~OpenGL, 100]:leave: ~OpenGL
    [opengl.cc, ~OpenGL, 87]:enter: ~OpenGL program: 1
    [opengl.cc, ~OpenGL, 100]:leave: ~OpenGL
    [image_process.cc, ClearEffect, 164]:enter: ClearEffect action_size: 0
    [image_process.cc, ClearEffect, 175]:leave: ClearEffect
    [camera_record.cc, DestroyPreviewSurface, 638]:enter DestroyPreviewSurface
    [camera_record.cc, onSurfaceDestroy, 436]:enter onSurfaceDestroy
I/System.out: onSurfaceDestroy
I/trinity: [camera_record.cc, onSurfaceDestroy, 456]:leave onSurfaceDestroy
I/trinity: [camera_record.cc, DestroyPreviewSurface, 649]:leave DestroyPreviewSurface
    [camera_record.cc, ReleaseCamera, 485]:enter ReleaseCamera
    [, , 0]:enter releaseCameraFromNative
    [, , 0]:leave releaseCameraFromNative
    [camera_record.cc, ReleaseCamera, 506]:leave ReleaseCamera
    [egl_core.cc, GetContext, 55]:return EGLCore GetContext...
    [camera_record.cc, Destroy, 597]:destroy context: 0xd9520ac0
I/trinity: [egl_core.cc, Release, 39]:after eglMakeCurrent...
    [egl_core.cc, Release, 41]:after eglDestroyContext...
    [camera_record.cc, Destroy, 606]:leave Destroy
E/trinity: [camera_record.cc, ProcessMessage, 523]:enter: ProcessMessage MESSAGE_QUEUE_LOOP_QUIT_FLAG
I/trinity: [camera_record.cc, ProcessMessage, 532]:leave ProcessMessage
I/trinity: [message_queue.cc, ~MessageQueue, 61]:CameraRecord message queue ~PacketQueue ....
    [message_queue.cc, Flush, 75]:
     CameraRecord message queue Flush .... and this time the queue_ Size is 0 
    [camera_record.cc, DestroyEGLContext, 206]:DestroyEGLContext leave
    [buffer_pool.h, ~BufferPool, 37]:enter: ~BufferPool size: 2
    [buffer_pool.h, ~BufferPool, 42]:leave: ~BufferPool

**Basically it just crashes randomly when we scroll fast backwards and forwards on that edit video with effects screen.**

**Comment by wlanjie** [comment]
logfile file, not logcat log

**Comment by wlanjie** [comment]
And the crash log, without this information I cannot locate you

**Comment by wlanjie** [comment]
![image](https://user-images.githubusercontent.com/4112636/94331253-8fa53880-fffd-11ea-9d1a-3d2a67762702.png)
[trinity_20200926.xlog.log](https://github.com/wlanjie/trinity/files/5286234/trinity_20200926.xlog.log)


Like this kind of log, if not, I can't confirm the problem for you

**Comment by arbaaj98** [comment]
I am facing same issue
